batch/batch/driver/instance_collection/pool.py

```diff
@@ -603,51 +611,62 @@ async def schedule_loop_body(self):
         }
 
         async def user_runnable_jobs(user):
-            async for batch in self.db.select_and_fetchall(
+            async for job_group in self.db.select_and_fetchall(
                 """
-SELECT batches.id, job_groups_cancelled.id IS NOT NULL AS cancelled, userdata, user, format_version
-FROM batches
-LEFT JOIN job_groups_cancelled
-       ON batches.id = job_groups_cancelled.id
-WHERE user = %s AND `state` = 'running';
+SELECT job_groups.batch_id, job_groups.job_group_id, t.cancelled IS NOT NULL AS cancelled, userdata, job_groups.user, format_version
+FROM job_groups
+LEFT JOIN batches ON job_groups.batch_id = batches.id
+LEFT JOIN LATERAL (
+  SELECT 1 AS cancelled
+  FROM job_group_self_and_ancestors
+  INNER JOIN job_groups_cancelled
+    ON job_group_self_and_ancestors.batch_id = job_groups_cancelled.id AND
+      job_group_self_and_ancestors.ancestor_id = job_groups_cancelled.job_group_id
+  WHERE job_groups.batch_id = job_group_self_and_ancestors.batch_id AND
+    job_groups.job_group_id = job_group_self_and_ancestors.job_group_id
+) AS t ON TRUE
```

Comment:
Same comment.

Reply:
See https://github.com/hail-is/hail/pull/14282#discussion_r1487618770